framework/decode/dx12_dump_resources.cpp

```diff
@@ -1421,10 +1431,10 @@ void Dx12DumpResources::WriteResource(nlohmann::ordered_json&   jdata,
         return;
     }
 
-    std::string file_name = prefix_file_name + "_res_id_" + std::to_string(resource_data->source_resource_id);
+    std::string file_name = prefix_file_name + "res_id_" + std::to_string(resource_data->source_resource_id);
```

Comment:
I think the underscore should not be removed. Filenames should be:
`...shader_resource_res_id_...`
instead of
`...shader_resourceres_id_...`

Same for the other places `_res_id_` was changed to `res_id_`

Reply:
But sometimes prefix_file_name is null, resulting in two underscores before res_id in the filename. Should I check for a null prefix_file_name?

Reply:
That value shouldn't be null. I have a bug fix that builds on this PR that should resolve that so we shouldn't get the double underscore in the name.

Reply:
Not sure it this change is the cause, but I'm getting file names with consecutive underscores, i.e. D3D12Multithreading.exe_trim_trigger_20240423T120337_dr__res_id_20_sub_0_.bin

Reply:
It was an existing problem. It also resulted in some missing files in the json and/or missing dumped .bin files.